src/providers/user.provider.tsx

```diff
@@ -22,26 +36,69 @@ export interface IProps {
 
 const UserProvider = ({children}: IProps) => {
   const [user, setUser] = useState<Patient | undefined>(undefined)
-  const [bloodPressures, setBloodPressures] = useState<BloodPressure[]>([])
+  const [bloodPressures, setBloodPressures] = useState<
+    BloodPressure[] | undefined
+  >(undefined)
+
+  const [medications, setMedications] = useState<Medication[] | undefined>(
+    undefined,
+  )
 
   const updatePatientData = async (
     patientResponseData: PatientResponseData,
   ) => {
-    // Todo - write/update patient offline
-    // Todo - write/update blood pressures offline
     const {
       patient_id,
       full_name,
       password_digest,
       blood_pressures,
+      medications,
     } = patientResponseData
-    setUser({patient_id, full_name, password_digest})
-    setBloodPressures(blood_pressures.slice())
+    const userData = {patient_id, full_name, password_digest}
+    const bloodPressuresData = blood_pressures.slice()
+    const medicationsData = (medications ?? []).slice()
+    setUser(userData)
+    setBloodPressures(bloodPressuresData)
+    setMedications(medicationsData)
+
+    try {
+      writeItemToDisk(userData, KEYS.USER)
+      writeItemsToDisk(bloodPressuresData, KEYS.BLOOD_PRESSURES)
+      writeItemsToDisk(medicationsData, KEYS.MEDICATIONS)
```

Comment:
as as `AsyncStorage` uses promises, I would have expected maybe these functions to be a promise which returned the `AsyncStorage.setItem`, then these 3 function calls could have been.

```
return Promises.all([writeItemToDisk(userData, KEYS.USER), writeItemsToDisk(bloodPressuresData, KEYS.BLOOD_PRESSURES), writeItemsToDisk(medicationsData, KEYS.MEDICATIONS)])
```

To know when it's done. Chaining promises seems to read a little nicer to me, than returning true at the end of async.